ui/components/resolvers/ResolversContainer.js

```diff
@@ -25,13 +25,19 @@ class ResolversContainer extends Component {
         return [];
     }
 
-    onResolverClicked(resolver) {
-        const { isResolverSaved } = this.state;
+    onResolverClicked(clickedResolver) {
+        const { resolver, isResolverSaved } = this.state;
+
+        if (resolver !== null
+            && resolver.type === clickedResolver.type
+            && resolver.field === clickedResolver.field) {
+            return;
+        }
 
         if (isResolverSaved) {
-            this.setState({ resolver });
+            this.setState({ resolver: clickedResolver });
         } else {
-            this.setState({ showChangeConfirmation: true, clickedResolver: resolver });
+            this.setState({ showChangeConfirmation: true, clickedResolver });
```

Comment:
So the state has two properties for the resolver, `resolver` and `clickedResolver`. Is that intentional?

Reply:
That is indeed intentional. `clickedResolver` is a temporal variable while the confirmation dialog is opened. In case the user cancels, it gets deleted but if they confirm, then `resolver = clickedResolver`.
In other words, `this.state.resolver` is the one controlling `ResolverDetail`.